lib/procinfo.cpp

```diff
@@ -135,10 +135,34 @@ void procinfo_non_boinc(PROCINFO& procinfo, PROC_MAP& pm) {
         procinfo.working_set_size += p.working_set_size;
     }
 #if 0
-    fprintf(stderr, "total non-boinc: %f %f\n", procinfo.user_time, procinfo.kernel_time);
+    fprintf(stderr,
+        "total non-boinc: %f %f\n", procinfo.user_time, procinfo.kernel_time
+    );
 #endif
 }
 
+// get CPU time of BOINC-related processes, low-priority processes,
+// and (if we're using Vbox) the Vbox daemon.
+//
+double boinc_related_cpu_time(PROC_MAP& pm, bool using_vbox) {
+    double sum = 0;
+    PROC_MAP::iterator i;
+    for (i=pm.begin(); i!=pm.end(); ++i) {
+        PROCINFO& p = i->second;
+#ifdef _WIN32
+        if (p.id == 0) continue;    // idle process
+#endif
+        if (
+            p.is_boinc_app
+            || p.is_low_priority
```

Comment:
The following is on indirectly related to this PR.

 `is_low_priority()` currently only does consider the "lowest-priority" nice value (unless I am mistaken). Once could argue that all "low-priority" nice values should be considered.

In any case, `is_low_priority()` should also return true for processes running under `SCHED_IDLE`.

Reply:
Maybe do this later.  It would require calling sched_getscheduler() for every process, every 10 sec.